scrollingpagerindicator/src/main/java/ru/tinkoff/scrollingpagerindicator/ScrollingPagerIndicator.java

```diff
@@ -432,12 +436,16 @@ protected void onDraw(Canvas canvas) {
 
                     if (dot - visibleFramePosition < currentScaleDistance) {
                         float calculatedDiameter = diameter * (dot - visibleFramePosition) / currentScaleDistance;
-                        if (calculatedDiameter < diameter) {
+                        if (dotMinimumSize > -1) {
+                            diameter = dotMinimumSize;
+                        } else if (calculatedDiameter < diameter) {
                             diameter = calculatedDiameter;
                         }
                     } else if (dot - visibleFramePosition > canvas.getWidth() - currentScaleDistance) {
                         float calculatedDiameter = diameter * (-dot + visibleFramePosition + canvas.getWidth()) / currentScaleDistance;
-                        if (calculatedDiameter < diameter) {
+                        if (dotMinimumSize > -1) {
+                            diameter = dotMinimumSize;
```

Comment:
You should calculate it based on current position to have smooth animation.
Something like on the line 445